architecture-building-systems/CityEnergyAnalyst
# Update photovoltaic.py

This PR resolves #3022 

Two parts are updated:
- calculate the angle of incidence, teta_rad, with pvlib
- limit air mass modifier to a feasible range (0.001,1). according to http://www.sciencedirect.com/science/article/pii/S0038092X05002410. 
![image](https://user-images.githubusercontent.com/16998305/134520602-7d55d3ce-fd31-4ff0-8400-b6ffcf1480d4.png)

From my side, the efficiencies for PV1 are around 10% on rooftops in summer.
![image](https://user-images.githubusercontent.com/16998305/134522552-4ac7afc7-a6c7-4cc3-8f3e-8020321ee209.png)
The efficiency peaks in the morning hours could be avoided by increase the lower bound of the air mass modifier.
Any recommendation is welcome!

*To test*
1. switch to branch i3022
2. in cea console, install pvlib by typing: `pip install pvlib`
3. choose your case study and run PV calculation
4. verify the results so we could discuss in the comments below

I am assigning @jimenofonseca  and @justinfmccarty as reviewers since you are both analysing results from this script.
